### PR TITLE
IL-282: Reset Password with email confirmation

### PR DESCRIPTION
**Pull request by nnick44**
@Sunagatov 
@yevr19

### CODE REVIEW

File: src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java [hunk 1]
```diff
@@ -120,4 +127,28 @@ public ResponseEntity<Void> deleteUserAvatar() {
         log.info("The request to delete the user avatar was handled.");
         return ResponseEntity.status(HttpStatus.OK).build();
     }
+
+    @Override
+    @PostMapping(path = "/password/reset")
+    public ResponseEntity<Void> resetUserPassword() {
+        UserDto userDto = securityPrincipalProvider.get();
+        log.info("Received the request to reset password the User with userId - {}.", userDto.getId());
```
Comment (annstriganova) [inline src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java:135]: nit: "for user with id: {}"

File: src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java [hunk 2]
```diff
@@ -120,4 +127,28 @@ public ResponseEntity<Void> deleteUserAvatar() {
...
+        return ResponseEntity.status(HttpStatus.OK)
+                .build();
+    }
+
+    @Override
+    @PostMapping(path = "/password/reset/confirm")
+    public ResponseEntity<Void> confirmResetUserPassword(@RequestBody final ConfirmEmailRequest confirmEmailRequest) {
+        log.info("Received email confirmation request to reset password");
+        emailTokenConformer.confirmResetPasswordEmailByCode(
+                new com.zufar.icedlatte.security.dto.ConfirmEmailRequest(confirmEmailRequest.getToken()));
```
Comment (annstriganova) [inline src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java:149]: nit: is it possible to replace FQN with just the method's name?

Comment (annstriganova) [reply src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java:149]: nit: oh, it seems to be conflicting with Open API's class. But what is the purpose of using both `record` and generated class?

**Review by annstriganova — APPROVED** [review]
Approved

File: src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java (hunk 1 above)
Comment (nnick44) [reply src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java]: fixed

File: src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java (hunk 2 above)
Comment (nnick44) [reply src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java:149]: fixed

Comment (annstriganova) [reply src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java:149]: This change is useful as well, but I was talking about this piece of code 😊

```java
// FQN (fully qualified name)
new com.zufar.icedlatte.security.dto.ConfirmEmailRequest(confirmEmailRequest.getToken()));
// VS simple name
new ConfirmEmailRequest(confirmEmailRequest.getToken()));
```

Comment (nnick44) [reply src/main/java/com/zufar/icedlatte/user/endpoint/UserEndpoint.java:149]: oh no:)
reverted

**Review by Sunagatov — APPROVED** [review]
Approved

**Pull request merged**